crates/findex/src/cloud.rs

```diff
@@ -465,63 +466,64 @@ impl FindexCallbacks<FindexCloudError, UID_LENGTH> for FindexCloud {
         Ok(true)
     }
 
-    async fn fetch_all_entry_table_uids(
-        &self,
-    ) -> Result<HashSet<Uid<UID_LENGTH>>, FindexCloudError> {
+    async fn fetch_all_entry_table_uids(&self) -> Result<Uids<UID_LENGTH>, FindexCloudError> {
         Err(FindexCloudError::Callback {
             error: "fetch all entry table uids not implemented in WASM".to_string(),
         })
     }
 
     async fn fetch_entry_table(
         &self,
-        entry_table_uids: HashSet<Uid<UID_LENGTH>>,
-    ) -> Result<Vec<(Uid<UID_LENGTH>, Vec<u8>)>, FindexCloudError> {
-        let serialized_uids = serialize_set(&entry_table_uids)?;
+        entry_table_uids: Uids<UID_LENGTH>,
+    ) -> Result<EncryptedMultiTable<UID_LENGTH>, FindexCloudError> {
+        let serialized_uids = serialize_set(&entry_table_uids.0)?;
 
         let bytes = self.post(Callback::FetchEntries, serialized_uids).await?;
 
-        Ok(deserialize_fetch_entry_table_results(&bytes)?)
+        Ok(EncryptedMultiTable(deserialize_fetch_entry_table_results(
+            &bytes,
+        )?))
     }
 
     async fn fetch_chain_table(
         &self,
-        chain_table_uids: HashSet<Uid<UID_LENGTH>>,
+        chain_table_uids: Uids<UID_LENGTH>,
     ) -> Result<EncryptedTable<UID_LENGTH>, FindexCloudError> {
-        let serialized_uids = serialize_set(&chain_table_uids)?;
+        let serialized_uids = serialize_set(&chain_table_uids.0)?;
 
         let bytes = self.post(Callback::FetchChains, serialized_uids).await?;
 
-        EncryptedTable::try_from_bytes(&bytes).map_err(FindexCloudError::from)
+        EncryptedTable::deserialize(&bytes).map_err(FindexCloudError::from)
     }
 
     async fn upsert_entry_table(
         &mut self,
         items: UpsertData<UID_LENGTH>,
     ) -> Result<EncryptedTable<UID_LENGTH>, FindexCloudError> {
-        let serialized_upsert = items.try_to_bytes()?;
+        let serialized_upsert = items.serialize()?;
 
         let bytes = self
-            .post(Callback::UpsertEntries, serialized_upsert)
+            .post(Callback::UpsertEntries, serialized_upsert.to_vec())
             .await?;
 
-        EncryptedTable::try_from_bytes(&bytes).map_err(FindexCloudError::from)
+        EncryptedTable::deserialize(&bytes).map_err(FindexCloudError::from)
     }
 
     async fn insert_chain_table(
         &mut self,
         items: EncryptedTable<UID_LENGTH>,
     ) -> Result<(), FindexCloudError> {
-        let serialized_insert = items.try_to_bytes()?;
+        let serialized_insert = items.serialize()?;
 
-        self.post(Callback::InsertChains, serialized_insert).await?;
+        self.post(Callback::InsertChains, serialized_insert.to_vec())
+            .await?;
```

Comment:
I think this is actually wrong to implement Serializable so that a `Zeroizing<Vec<u8>>` is returned since it implies cloning a lot of data...

Reply:
I guess this will be it for this version but we need to spend some more time on the subject.

Reply:
Did you benchmark this MR?

Reply:
actually benchmarks only deal with FPE scheme...

Reply:
We need a solid benchmark suite for CloudproofRust. But this would be done from the client I guess. I think Thibaud did something with the JS client.